test/e2e/basic/netdev_set_channels.go

```diff
@@ -0,0 +1,163 @@
+	util "github.com/openshift/cluster-node-tuning-operator/test/e2e/util"
+)
+
+// Test the tuned daemon functionality to adjust netdev queue count for physical network devices via ethtool.
+var _ = ginkgo.Describe("[basic][netdev_set_channels] Node Tuning Operator adjust netdev queue count", func() {
+	const (
+		profileNetdev   = "../testing_manifests/netdev_set_channels.yaml"
+		nodeLabelNetdev = "tuned.openshift.io/netdev-set-queue-count"
+	)
+
+	ginkgo.Context("adjust netdev queue count", func() {
+		var (
+			node *coreapi.Node
+		)
+
+		// Cleanup code to roll back cluster changes done by this test even if it fails in the middle of ginkgo.It()
+		ginkgo.AfterEach(func() {
+			ginkgo.By("cluster changes rollback")
+			if node != nil {
+				util.ExecAndLogCommand("oc", "label", "node", "--overwrite", node.Name, nodeLabelNetdev+"-")
+			}
+			util.ExecAndLogCommand("oc", "delete", "-n", ntoconfig.OperatorNamespace(), "-f", profileNetdev)
+		})
+
+		ginkgo.It("adjust netdev queue count for physical network devices via ethtool", func() {
+			var (
+				phyDev  string
+				explain string
+			)
+			const (
+				pollInterval = 5 * time.Second
+				waitDuration = 5 * time.Minute
+				chLen        = 2 // the number of multi-purpose channels to set by profileNetdev
+			)
+			cmdGetPhysicalDevices := []string{"find", "/sys/class/net", "-type", "l", "-not", "-lname", "*virtual*", "-printf", "%f "}
+
+			ginkgo.By("getting a list of worker nodes")
+			nodes, err := util.GetNodesByRole(cs, "worker")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(len(nodes)).NotTo(gomega.BeZero(), "number of worker nodes is 0")
+
+			node = &nodes[0]
+			ginkgo.By(fmt.Sprintf("getting a tuned Pod running on node %s", node.Name))
+			pod, err := util.GetTunedForNode(cs, node)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			util.Logf("found Pod %s running on node %s", pod.Name, node.Name)
+
+			ginkgo.By(fmt.Sprintf("getting a list of physical network devices: %v", cmdGetPhysicalDevices))
+			phyDevs, err := util.ExecCmdInPod(pod, cmdGetPhysicalDevices...)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			for _, d := range strings.Split(phyDevs, " ") {
+				if d == "" {
+					continue
+				}
+				// See if the device 'd' supports querying the channels.
+				_, err := util.ExecCmdInPod(pod, "ethtool", "-l", d)
+				if err == nil {
+					// Found a device that supports querying the channels
+					phyDev = d
+					break
+				}
+			}
+
+			if phyDev == "" {
+				util.Logf(fmt.Sprintf("no network devices supporting querying channels found on node %s; skipping test", node.Name))
+				return
+			}
+
+			cmdCombinedChannelsCurrent := []string{"bash", "-c", "ethtool -l " + phyDev + "| sed -n '/Current hardware settings/,/Combined:/{s/^Combined:\\s*//p}'"}
+			cmdCombinedChannelsMax := []string{"bash", "-c", "ethtool -l " + phyDev + "| sed -n '/Pre-set maximums/,/Combined:/{s/^Combined:\\s*//p}'"}
```

Comment:
one thing I like to do when doing such parsing is to add a comment showing how the text should look like
none of my devices support `ethtool -l`, so I can't double check what's happening here 
something like:
```
     /* sample output of `ethertool -l`:
     ..
     .. only what's relevant ..
     ..
     */
```
(just my 2c)

Reply:
and, in the lines below you do:
```
_, err = util.ExecCmdInPod(pod, "bash", "-c", fmt.Sprintf("ethtool -L %s combined 1", phyDev))
```
which seems a bit easier to read than `"ethtool -l " + phyDev + "...`, perhaps you could unify the string processing, and maybe split the long lines above to help the readability (new line after `-c` and before the `|`)

Reply:
Agreed, will fix.